main/src/main/java/org/mobilitydata/gtfsvalidator/validator/FeedServiceDateValidator.java

```diff
@@ -30,25 +28,16 @@
  * <p>Generated notice: {@link MissingFeedInfoDateNotice}.
  */
 @GtfsValidator
-public class FeedServiceDateValidator extends FileValidator {
-
-  private final GtfsFeedInfoTableContainer feedInfoTable;
-
-  @Inject
-  FeedServiceDateValidator(GtfsFeedInfoTableContainer feedInfoTable) {
-    this.feedInfoTable = feedInfoTable;
-  }
+public class FeedServiceDateValidator extends SingleEntityValidator<GtfsFeedInfo> {
 
   @Override
-  public void validate(NoticeContainer noticeContainer) {
-    for (GtfsFeedInfo feedInfo : feedInfoTable.getEntities()) {
-      if (feedInfo.hasFeedStartDate() && !feedInfo.hasFeedEndDate()) {
-        noticeContainer.addValidationNotice(
-            new MissingFeedInfoDateNotice(feedInfo.csvRowNumber(), "feed_end_date"));
-      } else if (!feedInfo.hasFeedStartDate() && feedInfo.hasFeedEndDate()) {
-        noticeContainer.addValidationNotice(
-            new MissingFeedInfoDateNotice(feedInfo.csvRowNumber(), "feed_start_date"));
-      }
+  public void validate(GtfsFeedInfo feedInfo, NoticeContainer noticeContainer) {
+    if (feedInfo.hasFeedStartDate() && !feedInfo.hasFeedEndDate()) {
+      noticeContainer.addValidationNotice(
+          new MissingFeedInfoDateNotice(feedInfo.csvRowNumber(), "feed_end_date"));
```

Comment:
And now I started thinking of an annotation for conditionally required fields that depend on presence of other fields...

Reply:
That is a good idea! It would allow to have more straightforward code and less validation code to be written by users! @barbeau what do ou think about that?

Reply:
Yeah, I was thinking the same thing. We already have the `@ConditionallyRequired` annotation so I suppose the implementation could just extend that? Although looking at the schema now `feedStartDate` and `feedEndDate` aren't annotated with this - probably because it's a WARNING and not an ERROR (i.e., the spec doesn't require it). I suppose we'd need to allow the annotation to specify if the corresponding notice would be an ERROR or WARNING?

Here are some other fields that are already annotated with `@ConditionallyRequired`:
* agency.agency_id
* stop_times.arrival_time
* stop_times.departure_time
* calendar_dates.txt (entire schema)
* calendar.txt (entire schema)
* fare_attributes.agency_id
* routes.agency_id
* routes.short_name
* routes.long_name
* stops.name
* stops.lat
* stops.lon
* stops.zone_id
* stops.parent_station
* ...

Some of the above are conditionally required depending on the *value* not the *presence* of other fields - we would just limit this annotation to generating rules for the *presence* of fields, right?